### ISSUE TITLE
Amounts paid and due

### ISSUE THREAD

**Issue by peacog** [National-Forestry-Authority/farmers#119]
**Amount to be paid**
Farmer reference
Year
Area allocated in hectares
Total due - calculate from area * land rent for the CFR (Land rent rates content type)
Total paid - calculate from payments
Outstanding amount - calculated = due - paid

Initial fields
Amount assessed - the is current amount owed by farmer - initial setting - this is first Amount due in table
Amount paid - 

List amounts 

System will compute amounts at start of year - Jan 1. 
Also allow administrator to request the recalculation- button/form in admin section

**Comment by peacog** [comment]
Computing the annual charges fails on staging with a memory error

**Comment by ygoex** [comment]
Pending to fill the spreadsheet to upload the information in bulk.

**Comment by ygoex** [comment]
On admin page we need to be able to select the year (2004)

We would need be able to recalculate the charges for an individual farmer. We would need to add an autocomplete field on the same admin page.

**Comment by peacog** [comment]
Hi @Rashids2021 @LeScruf 
I've added the options we spoke about for calculating annual charges:

- You can select the year from a dropdown, starting at 2004
- You can select a farmer to do the calculations for that farmer only
- You can opt to recalculate existing charges. The default action is to NOT recalculate if annual charges already exist. If you check the recalculate option the existing charge records will be deleted and a new one will be created


![Image](https://user-images.githubusercontent.com/1312689/232786049-3134f5c4-1efe-4022-ac10-a7ec4beb69bf.png)



You can test it here https://farmers.stg.envs.utils.nfa.go.ug/admin/config/nfafmis/settings

Note that the system will generate a payment advice (invoice) for each annual charge created. This is a feature that already existed from before we started working on it.

**Comment by peacog** [comment]
Hi @Rashids2021 @LeScruf 

In staging I have generated all of the annual charges from 2004 to 2022. Please take a look when you have time.

**Comment by peacog** [comment]
@LeScruf @Rashids2021 
I'm moving this ticket back to In Progress after discussing the problem of how to deal with land rate prior to 2016. We agreed that the calculate annual charges button should only apply after 2024

**Comment by peacog** [comment]
Closing in favor of https://github.com/National-Forestry-Authority/farmers/issues/158
